### PR TITLE
Separate moveit_experimental packages

### PR DESCRIPTION
**Pull request by rhaschke**
This is a replacement for #1549, factoring out all packages in `moveit_experimental` as proper ROS packages. This allows to build them separately as intended by #1549.
Currently, only `jog_arm` is enabled, all other packages are catkin-ignored (some of them are not even catkinized yet). In contrast to #1549, this PR leaves all folders in their place. 
That is, `moveit_experimental` just acts as a container folder for these packages.
As we don't plan to release `moveit_experimenal` packages, there is no need to define a corresponding meta package.

- [x] This PR builds on top of #1605 and just adds 3c5f3eb. 
Thus this PR depends on the former being merged first.

### CODE REVIEW

File: moveit_experimental/jog_arm/config/ur5_simulated_config.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+command_out_topic: joint_group_position_controller/command
+# What type of topic does your robot driver expect?
+# Currently supported are std_msgs/Float64MultiArray (for ros_control JointGroupVelocityController)
+# or trajectory_msgs/JointTrajectory (for Universal Robots and other non-ros_control robots)
+command_out_type: std_msgs/Float64MultiArray
+# Can save some bandwidth as most robots only require positions or velocities
+publish_joint_positions: true
+publish_joint_velocities: false
+publish_joint_accelerations: false
+
```
Comment (AndyZe) [inline moveit_experimental/jog_arm/config/ur5_simulated_config.yaml]: Delete 2 lines here

File: moveit_experimental/jog_arm/CMakeLists.txt [hunk 2]
```diff
@@ -1,31 +1,68 @@
-set(JOG_ARM_NAME jog_arm)
+cmake_minimum_required(VERSION 2.8.3)
+project(jog_arm)
```
Comment (v4hn) [inline moveit_experimental/jog_arm/CMakeLists.txt]: Are we happy naming this `jog_arm` or should this have a `moveit` prefix?

Comment (rhaschke) [reply moveit_experimental/jog_arm/CMakeLists.txt]: I'm definitely agree to name this package `moveit_jog_arm` in order to distinguish it from the existing `jog_arm` package. Actually, I wanted to ask @AndyZe to apply this change after the present PR is merged ;-)
Do you think, it should be part of the present PR instead?

Comment (AndyZe) [reply moveit_experimental/jog_arm/CMakeLists.txt]: Well, I like the shorter name.

`jog_arm` was never released as a debian for Melodic so there won't be a conflict there (and I don't mind removing the Kinetic debian from rosdistro). So don't worry about name conflicts.

Comment (AndyZe) [reply moveit_experimental/jog_arm/CMakeLists.txt]: Made a PR to remove the old version of `jog_arm` from rosdistro here:

https://github.com/ros/rosdistro/pull/21987

Comment (rhaschke) [reply moveit_experimental/jog_arm/CMakeLists.txt]: All packages of the MoveIt universe are prefixed with `moveit_`. This should hold for `jog_arm` as well. That's the "downside" of your decision to migrate the package into MoveIt :wink:

**Comment by v4hn** [comment]
You're right, renaming does not have to be part of this request.
I created a new issue for it here: #1612 .

@rhaschke , you mentioned you merge and push pull-requests almost exclusively on the command line. I just tried to squash your commits and merge (granted, this triggers the old squash-vs-plain-merge debate) from my system (opposed to the github web page), but the repository setup does not let me push, saying:
```
remote: error: GH006: Protected branch update failed for refs/heads/master.
remote: error: Required status check "continuous-integration/travis-ci" is expected. At least 1 approving review is required by reviewers with write access.
To github.com:ros-planning/moveit.git
 ! [remote rejected]     master -> master (protected branch hook declined)
error: failed to push some refs to 'git@github.com:ros-planning/moveit.git'
```

How do you handle this in your review process?

I am rather unhappy with safety measures that make it impossible to work with git the conventional way, @davetcoleman .

**Comment by rhaschke** [comment]
@v4hn, I perform the _review process_ locally, but usually, I use github's web interface to finally merge.
Not doing so, the manual commits would not (automatically) refer to github's pull request ID.

File: moveit_experimental/jog_arm/CMakeLists.txt (hunk 2 above)
Comment (davetcoleman) [reply moveit_experimental/jog_arm/CMakeLists.txt]: I like ``moveit_jog_arm``

The upside is we have really good tools in the moveit repo for code quality and maintenance :-)

**Comment by davetcoleman** [comment]
image

;-)

**Pull request merged**